trace/backend_collector.go

```diff
@@ -167,15 +168,19 @@ func NewBackendCollector(
 	packetCounts PacketCountConsumer,
 	sendWitnessPayloads bool,
 	plugins []plugin.AkitaPlugin,
-) Collector {
+) (Collector, error) {
+	redactor, err := data_masks.NewRedactor(svc, lc)
```

Comment:
Does this result in one goroutine updating the redaction list, for each network interface?  (It was already pretty silly that we had separate upload buffers and timers for each interface.)   I think we could easily move the redactor up a level out of the per-interface loop, and probably should. 

I don't think we can do the same with the BackendCollector itself as its Process() method is probably not safe against concurrent calls.

Reply:
In practice, most traffic is on a single interface anyway (we don't get a lot of actual parallelism ) So I thought for a bit about whether this would introduce blocking behavior where it previously was unnecessary and concluded that it probably would not.

An alternative model is using an array of channels to transmit the new version of the data into each goroutine, but that seems complex and (given the lack of actual parallelism, and the rarity of updates) slower than the read-lock.

Reply:
In 072366b.